packages/ember-glimmer/tests/integration/helpers/custom-helper-test.js

```diff
@@ -22,4 +26,431 @@ moduleFor('Helpers test: custom helpers', class extends RenderingTest {
+      },
+      compute() {
+        return ++count;
+      },
+      destroy() {
+        destroyCount++;
+        this._super();
+      }
+    });
+
+    this.render('{{hello-world "whut"}}');
+
+    this.assertText('1');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('1');
+
+    this.runTask(() => helper.recompute());
+
+    this.assertText('2');
+
+    assert.strictEqual(destroyCount, 0, 'destroy is not called on recomputation');
+  }
+
+  ['@htmlbars shorthand helper is called for param changes']() {
+    let computeCount = 0;
+
+    this.registerHelper('hello-world', ([value]) => {
+      computeCount++;
+      return `${value}-value`;
+    });
+
+    this.render('{{hello-world name}}', {
+      name: 'bob'
+    });
+
+    this.assertText('bob-value');
+
+    assert.strictEqual(computeCount, 1, 'compute is called exactly 1 time');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('bob-value');
+
+    assert.strictEqual(computeCount, 1, 'compute is called exactly 1 time');
+
+    this.runTask(() => set(this.context, 'name', 'sal'));
+
+    this.assertText('sal-value');
+
+    assert.strictEqual(computeCount, 2, 'compute is called exactly 2 times');
+
+    this.runTask(() => set(this.context, 'name', 'bob'));
+
+    this.assertText('bob-value');
+  }
+
+  ['@htmlbars helper compute is called for param changes']() {
+    let createCount = 0;
+    let computeCount = 0;
+
+    this.registerHelper('hello-world', {
+      init() {
+        this._super(...arguments);
+        // FIXME: Ideally, the helper instance does not need to be recreated
+        // for change of params.
+        createCount++;
+      },
+      compute([value]) {
+        computeCount++;
+        return `${value}-value`;
+      }
+    });
+
+    this.render('{{hello-world name}}', {
+      name: 'bob'
+    });
+
+    this.assertText('bob-value');
+
+    assert.strictEqual(computeCount, 1, 'compute is called exactly 1 time');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('bob-value');
+
+    assert.strictEqual(computeCount, 1, 'compute is called exactly 1 time');
+
+    this.runTask(() => set(this.context, 'name', 'sal'));
+
+    this.assertText('sal-value');
+
+    assert.strictEqual(computeCount, 2, 'compute is called exactly 2 times');
+
+    this.runTask(() => set(this.context, 'name', 'bob'));
+
+    this.assertText('bob-value');
+
+    assert.strictEqual(createCount, 1, 'helper is only created once');
+  }
+
+  ['@test shorthand helper receives params, hash']() {
+    let params, hash;
+
+    this.registerHelper('hello-world', (_params, _hash) => {
+      params = _params;
+      hash = _hash;
+    });
+
+    this.render('{{hello-world name "rich" last="sam"}}', {
+      name: 'bob'
+    });
+
+    assert.strictEqual(params[0], 'bob', 'first argument is bob');
+    assert.strictEqual(params[1], 'rich', 'second argument is rich');
+    assert.strictEqual(hash.last, 'sam', 'hash.last argument is sam');
```

Comment:
Can you somehow reflect these in the output and do `this.assertText` (or similar) instead?

Maybe something like: `params: ${JSON.stringify(_params)}, hash: ${JSON._hash)}`